src/meltano/core/plugin/singer/tap.py

```diff
@@ -615,3 +619,58 @@ def catalog_cache_key(self, plugin_invoker):
         key_json = json.dumps(key_dict)
 
         return sha1(key_json.encode()).hexdigest()  # noqa: S303 S324
+
+    def warn_property_not_found(  # noqa: C901
+        self,
+        rules: list[MetadataRule],
+        catalog: CatalogDict,
+    ):
+        """Validate MetadataRules conforms to discovered Catalog.
+
+        Validate MetadataRules against the tap's discovered Catalog & emit
+        warnings for each rule that does not match any property in catalog.
+
+        The method is intentionally written in a defensive manner so as not to
+        raise exceptions. e.g.
+            * use dict.get rather than []-accessor
+            * filter list comprehensions to remove some elements that should
+              not be used for validation
+
+        Args:
+            rules: List of `MetadataRule`
+            catalog: Discovered Source Catalog
+        """
+        property_rules = [
+            r
+            for r in rules
+            if r.tap_stream_id != "*"
```

Comment:
Meltano also accepts wildcards _within_ stream and property names, so I think this would   try to find a field `prefix_*` or a stream `public-*`, for example. We should probably account for that and include some test cases.

For implementation details: Meltano uses [fnmatch](https://docs.python.org/3/library/fnmatch.html).

Reply:
I am not sure if we can reliably emit a warning for wildcard matches within streams and property names.

for instance: 

if a tap has 3 streams `activity_account`, `activity_user`, `activity_role`, and the user specifies the following in the `meltano.yml` file:

```
extractors:
  - name: tap-example
    select:
      - activity_*.*
```

IMO this should not emit a warning, as `activity_*` is matched by all 3 streams

If a tap has 3 streams as above, and the first two have the field `email`, then I'm not sure what the expected behavior ought to be for the following select extra

```
extractors:
  - name: tap-example
    select:
      - activity_*.email
```

because the selection rule will match at least once, but will not match all the streams captured by `activity_*`. Maybe we could warn that `activity_role` does not have a field named `email`

If instead of a top level field named `email`, `activity_account` had a field called `primary_contact_email`, while `activity_user` had the field `email`, a user of meltano might specify this selection as 

```
extractors:
  - name: tap-example
    select:
      - activity_*.*email
```

I am leaning towards the notion that trying to handle wild-card matches might not be possible, and if a wildcard match is encountered, the method should log an info message verification of field existence was skipped as a result of wildcard matches.

I'd love to know your thoughts before proceeding

Reply:
Yes, I think this is the simpler approach. Changing.

Reply:
- updated the list comprehension filter to `"*" in r.tap_stream_id`
- changed the function definition as above.